feat(notifier): adds my purchases page

jurajpiar opened the pull request:
Resolving [RMKT-630](https://rsklabs.atlassian.net/browse/RMKT-630) and [RMKT-631](https://rsklabs.atlassian.net/browse/RMKT-631) by adding my purchases page and fancy expiry date

nduchak committed:
feat(notifier): adds my purchases page

feat: draft for cache notifier subscriptions service

jurajpiar committed:
feat(notifier): maps subscriptions to my purchases

jurajpiar committed:
feat(notfier): adds plan to subscriptions model

jurajpiar committed:
feat(notifier): adds and refactors tab header

jurajpiar committed:
feat(notifier): adds expiration date formatting

refactor(notifier): moves stuff around

+ stops disabling view button when renew is blcoked

jurajpiar commented on src/components/molecules/CombinedPriceCell.tsx:
```diff
@@ -9,15 +10,27 @@ export interface CombinedPriceCellProps {
+    },
+  },
+}))
+
```
maybe move to HOC folder.

jurajpiar commented on src/components/molecules/TabHeader.tsx:
```diff
@@ -0,0 +1,32 @@
+    <Grid container item xs={10} alignItems="center">
+      <Grid item xs={1}>
+        {icon}
+        {/* <img src={handProvidingFunds} alt="hand providing funds" /> */}
```
remove commented out code

jurajpiar commented on src/components/organisms/notifier/NotifierPlanDescription.tsx:
```diff
@@ -41,16 +41,16 @@ const NotifierPlanDescription: FC<Props> = ({ item, crypto, currentFiat }) => {
         </Grid>
         <Grid item xs={1} md={2}>
-          <Typography color="textPrimary">{item.item['token'].displayName}</Typography>
+          <Typography color="textPrimary">{item.item.token.displayName}</Typography>
```
use destructured properties

jurajpiar commented on src/components/organisms/notifier/NotifierPlanDescription.tsx:
```diff
@@ -41,16 +41,16 @@ const NotifierPlanDescription: FC<Props> = ({ item, crypto, currentFiat }) => {
+          <Typography color="textPrimary">{item.item.token.displayName}</Typography>
         </Grid>
         <Grid item xs={1} md={2}>
           <Typography color="textPrimary">
```
line below - extract to const

jurajpiar commented on src/components/pages/notifier/myoffers/mapActiveContracts.tsx:
```diff
@@ -41,38 +37,18 @@ const mapActiveContracts = (
         <Grid item>
           <Typography color="primary" variant="body2">
-            {parseToBigDecimal(price).mul(crypto?.[token.symbol]?.rate)?.toFixed(2)}
+            {price.mul(crypto?.[token.symbol]?.rate)?.toFixed(2)}
```
extract to const

jurajpiar commented on src/components/pages/notifier/mypurchase/NotifierMyPurchasePage.tsx:
```diff
@@ -0,0 +1,111 @@
+  const reportError = useErrorReporter()
+
+  const [subscriptions, setSubscriptions] = useState<Array<NotifierSubscriptionItem>>()
+  const [isTableLoading, setIsTableLoading] = useState<boolean>(false)
```
no need for defaulting this state

jurajpiar commented on src/components/pages/notifier/mypurchase/NotifierMyPurchasePage.tsx:
```diff
@@ -0,0 +1,111 @@
+      setIsTableLoading(true)
+      subscriptionsApi.connect(reportError)
+      subscriptionsApi.fetch({
+        // consumer: account,
```
uncomment

jurajpiar commented on src/components/pages/notifier/mypurchase/mapMyPurchases.tsx:
```diff
@@ -0,0 +1,87 @@
+import ExpirationDate, { SubscriptionExpirationType } from 'components/molecules/ExpirationDate'
+import { PlanDTO } from 'api/rif-marketplace-cache/notifier/offers/models'
+
+const EXPIRATION_WARNING_TRIGGER = 5
+
+const getExpirationType = (
+  { planStatus, daysLeft }: PlanDTO,
+): SubscriptionExpirationType => {
+  if (planStatus === 'INACTIVE' || daysLeft <= 0) return 'blocked'
+
+  if (daysLeft <= EXPIRATION_WARNING_TRIGGER) return 'warning'
+
+  return 'normal'
+}
```
move to utils

0xartem commented on src/api/rif-notifier-service/index.ts:
```diff
@@ -3,36 +3,100 @@ import { AbstractAPIService } from 'api/models/apiService'
+        })
+      }
+
+      const {
+        status,
+        content: {
+          apiKey,
+        },
+      } = await response.json()
```
I think this can go in one line

jurajpiar replied:
that won't fit in 80 characters

jurajpiar reviewed (commented):
Need to do some refactoring.

0xartem commented on src/api/rif-notifier-service/index.ts:
```diff
@@ -3,36 +3,100 @@ import { AbstractAPIService } from 'api/models/apiService'
+        })
+      }
+
+      const {
+        content,
+        message,
+      } = await response.json()
```
this as well

0xartem reviewed (approved):
looks good, does it involve changes from you and naz. or these only yours?

jurajpiar commented:
> does it involve changes from you and naz. or these only yours?

Both.

itofarina commented on src/context/Market/interfaces.ts:
```diff
@@ -6,11 +6,12 @@ export type MarketErrorId = 'market-init'
 export type ContextName = 'market'
-export type State = ContextState & {
-  exchangeRates: {
+export type ExchageRate = {
```
```suggestion
export type ExchangeRate = {
```

itofarina commented on src/hooks/useDisabledStyle.tsx:
```diff
@@ -0,0 +1,11 @@
+import { makeStyles, Theme } from '@material-ui/core'
+
+const useDisabledStyle = makeStyles(({ palette }: Theme) => ({
+  root: {
+    '& *': {
+      color: palette.text.disabled,
+    },
+  },
+}))
```
😍

itofarina commented on src/hooks/useDisabledStyle.tsx:
```diff
@@ -0,0 +1,11 @@
+import { makeStyles, Theme } from '@material-ui/core'
```
```suggestion
import { makeStyles, Theme } from '@material-ui/core/styles'
```
would minimize bundle size

itofarina commented on src/components/pages/notifier/mypurchase/mapMyPurchases.tsx:
```diff
@@ -0,0 +1,87 @@
+          {
+            disabled: expType === 'blocked',
+            id: `renew_${id}`,
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            handleSelect: (): void => {},
+            children: 'Renew',
+          }, {
+            id: `view_${id}`,
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            handleSelect: (): void => {},
```
May be better use the logger to print a message like "Not implemented yet"? So at least we know what's going on when clicking in dev mode

itofarina commented on src/components/pages/notifier/myoffers/mapActiveContracts.tsx:
```diff
@@ -83,26 +62,21 @@ const mapActiveContracts = (
+        actions={[
+          {
+            id: `withdraw_${customerId}`,
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            handleSelect: (): void => {},
+            children: 'Withdraw',
+          }, {
+            id: `view_${customerId}`,
+            // eslint-disable-next-line @typescript-eslint/no-empty-function
+            handleSelect: (): void => {},
```
May be better use the logger to print a message like "Not implemented yet"? So at least we know what's going on when clicking in dev mode

itofarina commented on src/components/pages/notifier/myoffers/mapActiveContracts.tsx:
```diff
@@ -1,16 +1,13 @@
-  Grid,
-} from '@material-ui/core'
+import React from 'react'
+import { Grid } from '@material-ui/core'
```
```suggestion
import Grid from '@material-ui/core/Grid'
```

itofarina commented on src/components/pages/notifier/myoffers/NotifierMyOffersPage.tsx:
```diff
@@ -121,9 +124,13 @@ const NotifierMyOffersPage: FC = () => {
     }
   }, [account, web3, reportError])
 
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
   const handleAddPlan = () => {}
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
   const handleEditPlan = () => {}
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
   const handleCancelPlan = () => {}
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
   const handleEditProfile = () => {}
```
May be better use the logger to print a message like "Not implemented yet"? So at least we know what's going on when clicking in dev mode. We would also avoid muting the linter here

itofarina commented on src/components/organisms/notifier/mypurchase/PurchasesTable.tsx:
```diff
@@ -0,0 +1,63 @@
+import {
+  makeStyles, TableContainer, Theme,
+} from '@material-ui/core'
```
```suggestion
import {
  makeStyles, Theme,
} from '@material-ui/core/styles'
import TableContainer from '@material-ui/core/TableContainer'
```

itofarina commented on src/components/organisms/notifier/NotifierPlanDescription.tsx:
```diff
@@ -16,9 +16,15 @@ type Props = {
+    provider, limit, channels,
+  } = item.item
+  const rate = crypto?.[token.symbol]?.rate || 0
+  const valueInFiat = ((value.mul(rate))).toFixed(2)
```
```suggestion
  const valueInFiat = (value.mul(rate)).toFixed(2)
```

itofarina commented on src/components/organisms/notifier/NotifierPlanDescription.tsx:
```diff
@@ -16,9 +16,15 @@ type Props = {
+  const {
+    token, daysLeft, value,
+    provider, limit, channels,
+  } = item.item
```
I might me missing something and may be it's not related to this PR but why is it nested?

jurajpiar replied:
Because the first item is meant to be order. I'll change it. Thanks for spotting that.

itofarina commented on src/components/molecules/TabHeader.tsx:
```diff
@@ -0,0 +1,31 @@
+import React, {
+  FC,
+} from 'react'
+import {
+  Grid, Typography,
+} from '@material-ui/core'
```
```suggestion
import Grid from '@material-ui/core/Grid'
import Typography from '@material-ui/core/Typography'
```

itofarina commented on src/components/molecules/ExpirationDate.tsx:
```diff
@@ -0,0 +1,54 @@
+import Typography from '@material-ui/core/Typography/Typography'
```
```suggestion
import Typography from '@material-ui/core/Typography'
```

itofarina commented on src/components/molecules/ExpirationDate.tsx:
```diff
@@ -0,0 +1,54 @@
+import Typography from '@material-ui/core/Typography/Typography'
+import React, { FC } from 'react'
+import { getShortDateString } from 'utils/dateUtils'
+import {
+  makeStyles, Theme, Grid,
+} from '@material-ui/core'
```
```suggestion
import {
  makeStyles, Theme, Grid,
} from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
```

itofarina reviewed (commented):
Sorry for too many comments, most of them are minor and overall looks great!

jurajpiar committed:
refactor(notiffier): fixes typos + imports

Update src/hooks/useDisabledStyle.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

Update src/components/organisms/notifier/NotifierPlanDescription.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

Update src/components/molecules/TabHeader.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

Update src/components/molecules/ExpirationDate.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

Update src/components/molecules/ExpirationDate.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

Update src/components/organisms/notifier/mypurchase/PurchasesTable.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

Update src/components/pages/notifier/myoffers/mapActiveContracts.tsx

Co-authored-by: Alvaro Fariña <farinalvaro@gmail.com>

jurajpiar committed:
refactor(notifier): refactors buy checkout page

jurajpiar committed:
feat(notiffier): adds logging of unimplememnted

jurajpiar commented:
> Sorry for too many comments, most of them are minor and overall looks great!

No mate! Thanks!
